apps/design/backend/src/app.ts

```diff
@@ -175,57 +170,63 @@ function buildApi({ translator, workspace }: AppContext) {
     async exportAllBallots(input: {
       electionId: Id;
     }): Promise<{ zipContents: Buffer; electionHash: string }> {
-      const {
-        ballotLanguageConfigs,
-        election,
-        layoutOptions,
-        nhCustomContent,
-      } = store.getElection(input.electionId);
+      const { election, ballotLanguageConfigs } = store.getElection(
+        input.electionId
+      );
 
       const zip = new JsZip();
 
-      let electionHash: string | undefined;
+      const renderer = await createPlaywrightRenderer();
```

Comment:
Thanks for digging into this. I was thinking that probably the entire app could just share one renderer that's created on startup, since it's still just being run locally. Would just need to figure out a good way to clean it up on exit. Definitely possible to move to a worker/pool model when we deploy VxDesign and start having concurrent users, but that may still be a long ways off, so I'm hesitant to invest in it prematurely.

As far as limiting the Chromium processes, my guess is that there's one for each `Page` in Playwright. We could likely change that by limiting how many pages we render concurrently (see https://github.com/votingworks/vxsuite/pull/4703/files#diff-87be57e5bc701f7f731a08da4c4002bc53fcf67df8f2911b74c207df1c14fa46R435)

Reply:
Thinking through this more, the structure of the rendering pipeline doesn't easily support limiting the number of Chromium pages open at a time, since we return all of the pages to the caller to allow further modification (e.g. marking the ballots, rendering to PDF, etc). It will take some more thinking if we want to change that, since all of the ballots need to be rendered first in order to extract the grid layouts and create the QR code. Maybe the answer is having some way to easily convert a Chromium page back into an HTML document so we can use a pool of Chromium pages like you suggested (likely trading off some speed for more conservation resource usage). Still leaning towards kicking this can down the line, unless you think there's a reason to prioritize it now.

Reply:
I don't feel strongly enough to block it. I'll just point out that spinning up as many `chromium` instances as there are ballot configurations is something that I cannot see persisting once we want this to run somewhere other than an engineer's machine. These things are almost easier to change *before* we build a whole lot of infrastructure on top of it. But I'm not in charge of the prioritization here, so I leave evaluating that tradeoff to others.

Reply:
Yes, I hear that. I'll merge for now just to avoid the rebase issues since I'm confident I can come back and address this before building on top of this more.